Add Javadoc for X-Pack SecurityClient

hmcc opened the pull request:
Connected to #30077.

I've added Javadoc for all the methods in X-Pack's `SecurityClient.java`.

When I was trying to find out more about how the methods were used, I realised that some of them *weren't* used. Rather than listing them out here, I'll add a second commit which removes them, which I'm very happy to revert.

elasticmachine commented:
Pinging @elastic/es-security

jaymode commented:
Hi @hmcc, thank you for your contribution. 

> When I was trying to find out more about how the methods were used, I realised that some of them weren't used.

Some of these methods are not used in our code base but are made available for our users. The SecurityClient can be used within custom java code so we provide methods for most APIs even if they are not used within our code base. If you do not mind reverting that commit, that would be great.

hmcc commented:
No problem, thanks for explaining!

jaymode commented on x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/client/SecurityClient.java:
```diff
@@ -103,24 +103,28 @@ public SecurityClient(ElasticsearchClient client) {
     /**
-     * Clears the realm caches. It's possible to clear all user entries from all realms in the cluster or alternatively
-     * select the realms (by their unique names) and/or users (by their usernames) that should be evicted.
+     * Populates a {@link ClearRealmCacheRequest} to clear the realm caches.
```
I would say that this comment applies generally to all of the changes to `prepare*` methods

hmcc replied:
Yes, agreed; that was only done for consistency with what was there before. I'll change the `prepare*` methods now.

jaymode replied:
You're absolutely correct that some have a different behavior; the javadocs for the change password method would be fine saying something like "Populates a change password request to change the password of the provided user".

hmcc commented:
@jaymode, I've updated the `prepare*` Javadocs where nothing is populated. I've left things alone where most/all of the request is populated, and I've replaced "populates" with "initializes" if some fields are populated but not others.

Is that OK?
